crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -5185,6 +5210,24 @@ impl<'db> TypeInferenceBuilder<'db> {
                 }
             };
 
+            let current_file = self.file();
+
+            let skip_non_global_scopes = !file_scope_id.is_global()
+                && symbol_table
+                    .symbol_id_by_name(symbol_name)
+                    .is_some_and(|symbol_id| {
+                        self.index
+                            .symbol_is_global_in_scope(symbol_id, file_scope_id)
+                    });
```

Comment:
This looks similar to something we do above in `add_binding`; we could extract a method?